### PR TITLE
libvirt: do not use lease as base domain

### PR DESCRIPTION
**Pull request by r4f4**
It causes problems with ocp-4.9 for containing `-`. Use a static, but
configurable, base domain instead.

Signed-off-by: Rafael Fonseca <r4f4rfs@gmail.com>

### CODE REVIEW

File: clusters/build-clusters/01_cluster/sshd-bastion/coredns/coredns_config.yaml [hunk 1]
```diff
@@ -8,11 +8,11 @@ data:
             rewrite stop name exact lnxocp01 sshd-0.bastion-z.svc.cluster.local
             rewrite stop name exact lnxocp02 sshd-1.bastion-z.svc.cluster.local
             rewrite stop name exact lnxocp03 sshd-2.bastion-z.svc.cluster.local
-            rewrite stop name regex ^.*libvirt-s390x-([0-9])-[0-9]\.$ sshd-{1}.bastion-z.svc.cluster.local
+            rewrite stop name regex ^.*libvirt-s390x-([0-9])-[0-9].* sshd-{1}.bastion-z.svc.cluster.local
```
Comment (deepsm007) [inline clusters/build-clusters/01_cluster/sshd-bastion/coredns/coredns_config.yaml:11]: To test we need to update coredns changes in separate PR and restart the pods

Comment (r4f4) [reply clusters/build-clusters/01_cluster/sshd-bastion/coredns/coredns_config.yaml:11]: Move it here https://github.com/openshift/release/pull/20350. I'll update this PR when that one merges.

**Review by andymcc — CHANGES_REQUESTED** [review]
We'll need a slightly different plan, since the CLUSTER_NAME is dynamic (because of the JOB_NAME_HASH) we won't be able to balance on the hostname in HAProxy - which we rely on quite heavily. We need the last portion (after the dynamic bits) to be static but represent an individual lease.
Perhaps we could set the baseDomain to be "$LEASED_RESOURCE.$BASE_DOMAIN" and remove the leased resource from the cluster_name. That way the end is predictable, and representative of the lease, but the last section of the hostname doesn't contain a "-".

**Pull request merged**